tests/rptest/tests/archive_retention_test.py

```diff
@@ -0,0 +1,124 @@
+
+    def produce(self, topic, msg_count):
+        msg_size = 1024 * 256
+        topic_name = topic.name
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       topic_name,
+                                       msg_size=msg_size,
+                                       msg_count=msg_count)
+
+        producer.start()
+        producer.wait()
+        producer.free()
+
+        def all_partitions_spilled():
+            return self.num_manifests_uploaded() > 0
+
+        wait_until(all_partitions_spilled, timeout_sec=180, backoff_sec=10)
+
+    def num_manifests_uploaded(self):
+        s = self.redpanda.metric_sum(
+            metric_name=
+            "redpanda_cloud_storage_spillover_manifest_uploads_total",
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+        self.logger.info(
+            f"redpanda_cloud_storage_spillover_manifest_uploads = {s}")
+        return s
+
+    def num_segments_deleted(self):
+        s = self.redpanda.metric_sum(
+            metric_name="redpanda_cloud_storage_deleted_segments_total",
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+        self.logger.info(f"redpanda_cloud_storage_deleted_segments_total = {s}")
+        return s
+
+    @cluster(num_nodes=4)
+    @matrix(cloud_storage_type=get_cloud_storage_type())
+    def test_delete_by_size(self, cloud_storage_type):
+        topic = TopicSpec(redpanda_remote_read=True,
+                          redpanda_remote_write=True,
+                          cleanup_policy=TopicSpec.CLEANUP_DELETE)
+
+        self.client().create_topic(topic)
+
+        def wait_for_topic():
+            wait_until(lambda: len(
+                list(self.client().describe_topic(topic.name))) > 0,
+                       30,
+                       backoff_sec=2)
+
+        def produce_until_spillover():
+            initial_uploaded = self.num_manifests_uploaded()
+
+            def new_manifest_spilled():
+                self.produce(topic, 250)
+                num_spilled = self.num_manifests_uploaded()
+                return num_spilled > initial_uploaded
+
+            wait_until(new_manifest_spilled,
+                       timeout_sec=120,
+                       backoff_sec=2,
+                       err_msg="Manifests were not created")
+
+        wait_for_topic()
+        produce_until_spillover()
+
+        # Enable cloud retention for the topic to force deleting data
+        # from the 'archive'
+        wait_for_topic()
+        self.client().alter_topic_config(topic.name, 'retention.bytes', 0x1000)
+        self.client().alter_topic_config(topic.name, 'retention.ms', -1)
+
+        wait_until(lambda: self.num_segments_deleted() > 50,
+                   timeout_sec=100,
+                   backoff_sec=5,
+                   err_msg=f"Segments were not removed from the cloud")
```

Comment:
I don't think the number of deleted segments is a great metric for retention application. There's partial uploads and segment jitter that can make this flaky. Would be great to use one of the cloud log size utils in `BucketView` to validate against the retention configuration.